src/main/java/core/basesyntax/strategy/BalanceOperation.java

```
@@ -0,0 +1,11 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.common.FruitTransaction;
+import core.basesyntax.db.Storage;
+
+public class BalanceOperation implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction fruitTransaction) {
```

The `apply` method does not validate its input. The task requires handling invalid inputs (including null parameters) and failing fast with informative exceptions: "Remember that your code will need to be tested, so try to anticipate and handle all invalid input data in advance. For example: ... 3. Null parameters" . Add a null-check for `fruitTransaction` and throw an informative RuntimeException if it is null.